firebase-database/src/main/java/com/google/firebase/database/core/utilities/Utilities.java

```diff
@@ -73,6 +81,13 @@ public static ParsedUrl parseUrl(String url) throws DatabaseException {
       originalPathString = originalPathString.replace("+", " ");
       Validation.validateRootPathString(originalPathString);
 
+      // TODO: Should log this out
+      if (serviceSettings != null) {
+        repoInfo.secure = false;
+        repoInfo.host = serviceSettings.host + ":" + serviceSettings.port;
+        repoInfo.internalHost = repoInfo.host;
```

Comment:
Should we add an explicit way to decide which credentials to use? I am asking because of Yuchen's feedback here: https://github.com/firebase/firebase-js-sdk/pull/3228

Reply:
I do think that would be a nice addition one day but this API was approved cross-platform for Android, iOS, and JS without any mention of a credentials setting so I think I'd need to do a new API review for that.

Do you anticipate bugs like that JS one in this implementation?

Reply:
This should be low risk since Android SDK is not reused in Admin SDKs. The expected behavior is the same as production -- get credentials from Firebase Auth if used, or no credentials if not.